### ISSUE TITLE
[BUG] RELATIVE refs could not be loaded

### ISSUE THREAD

**Issue by SimeonGerginov** [OpenAPITools/openapi-generator#9991]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected output?

##### Description

After the latest version of the **`OpenAPI Generator`** was released: **`5.2.0`**, a new bug was introduced regarding relative **`$refs`**: during generation the following error occurs:

```yaml
[main] WARN  io.swagger.v3.parser.OpenAPIV3Parser - Exception while resolving:
java.lang.RuntimeException: Unable to load RELATIVE ref: ./specs/my_spec.yaml path: /specs/spec_path
````

This behaviour is only seen in the latest released version, previously it was working correctly. I've looked through the **`CHANGELOG`** document but couldn't find an entry that could have introduced this problem.

##### openapi-generator version

5.2.0

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  title: MyExampleSpec
  description: >-
paths: {}
components:
  schemas:
    MyExampleSchema:
      description: >-
      properties:
        id:
          description: The id of MyExampleSchema.
          type: string
        message:
          description: The message of MyExampleSchema.
          $ref: ../specs/MySchemas.yaml#/components/schemas/Message
```

##### Steps to reproduce

1. Build the **`OpenAPI Generator`**.
2. Run one of the generators ( I've tested it with **`PowerShell`** and **`Python`**) with the provided **`yaml`** using the following command:

```yaml
java openapi-generator-cli-5.2.0.jar generate -g powershell -i "myOpenAPISpec.yaml" --global-property
``` 

3. Observe the errors that occur which should match the mentioned error message above.

##### Related issues/PRs

none

##### Suggest a fix

Currently no fix suggestion can be provided. I'll try to find the root cause of the issue and then think about what kind of fix can be added.

**Comment by SimeonGerginov** [comment]
I was able to track this bug to this [Pull Request](https://github.com/OpenAPITools/openapi-generator/pull/9631). **`v2.0.26`** of the **`swagger-parser`** causes this. I've tried with the previous version **`v2.0.25`** and everything works as expected. So the problem is present in **`v2.0.26`** and above as with the latest released version, this is still not fixed.

@wing328 do you have a process when an issue with a bumped dependency occurs ? Do you revert back to the latest working version or you provide a fix in the dependency package ?

**Comment by ydmitry** [comment]
Any news on this issue?

**Comment by gbaah** [comment]
I am hitting this issue. Has a fix been provided?

**Comment by equalspeterd** [comment]
It appears the underlying framework has addressed this issue: https://github.com/swagger-api/swagger-parser/pull/1557

**Comment by graham2071** [comment]
Correct, checked with openapitools-generator 6.0.0 and it's resolved.
